src/fldeff_dig.c

```diff
@@ -0,0 +1,49 @@
+#include "global.h"
+#include "field_effect.h"
+#include "field_player_avatar.h"
+#include "fldeff.h"
+#include "item_use.h"
+#include "overworld.h"
+#include "party_menu.h"
+#include "sprite.h"
+#include "constants/map_objects.h"
+
+static void hm2_dig(void);
+static void sub_80C9AFC(void);
+
+bool8 SetUpFieldMove_Dig(void)
+{
+    if (CanUseEscapeRopeOnCurrMap() == TRUE)
+    {
+        gFieldCallback2 = FieldCallback_PrepareFadeInFromMenu;
+        gPostMenuFieldCallback = hm2_dig;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+static void hm2_dig(void)
+{
+    Overworld_ResetStateAfterDigEscRope();
+    FieldEffectStart(FLDEFF_USE_DIG);
+    gFieldEffectArguments[0] = GetCursorSelectionMonId();
+}
+
+bool8 FldEff_UseDig(void)
+{
+    u8 taskId = oei_task_add();
+
+    gTasks[taskId].data[8] = (u32)sub_80C9AFC >> 16;
```

Comment:
The pointer storage here (and following line) is probably a macro expansion.

Reply:
Do you mean I should define a macro called `FLDEFF_SET_FUNCTION_TO_DATA(func)` or so?

Reply:
Yes.

Reply:
do you expect I to add a semicolon when using the macro so that it looks like a function?

Reply:
Yes, as it is stylistically consistent with the remainder of the repository.